File: android/src/main/kotlin/com/maktubcompany/easy_blue_printer/plugin/domain/usecase/PrintUseCase.kt
package com.maktubcompany.easy_blue_printer.plugin.domain.usecase

import com.maktubcompany.easy_blue_printer.plugin.domain.repository.BluetoothRepository

class PrintUseCase(private val repository: BluetoothRepository) {
    fun execute(data: String, size: Int, align: Int, bold: Boolean): Boolean {
        return repository.printData(data, size, align, bold)
    }
}

File: ios/easy_blue_printer/Sources/easy_blue_printer/plugin/domain/usecases/PrintUseCase.swift
import Foundation

public class PrintUseCase {
    private let repository: BluetoothRepository

    public init(repository: BluetoothRepository) {
        self.repository = repository
    }

    public func execute(data: String, size: Int, align: Int, bold: Bool) -> Bool {
        return repository.printData(data: data, size: size, align: align, bold: bold)
    }
}
